### PR TITLE
SOLR-10461: move setAliveCheckInterval from clients into Builder

### PR DESCRIPTION
**Pull request by epugh**
https://issues.apache.org/jira/browse/SOLR-10461

# Description

Move to using builders everywhere.
# Solution

move setAliveCheckInterval from client to Builder.

# Tests

Please describe the tests you've developed or run to confirm this patch implements the feature or solves the problem.

# Checklist

Please review the following and check all that apply:

- [ ] I have reviewed the guidelines for [How to Contribute](https://wiki.apache.org/solr/HowToContribute) and my code conforms to the standards described there to the best of my ability.
- [ ] I have created a Jira issue and added the issue ID to my pull request title.
- [ ] I have given Solr maintainers [access](https://help.github.com/en/articles/allowing-changes-to-a-pull-request-branch-created-from-a-fork) to contribute to my PR branch. (optional but recommended)
- [ ] I have developed this patch against the `main` branch.
- [ ] I have run `./gradlew check`.
- [ ] I have added tests for my changes.
- [ ] I have added documentation for the [Reference Guide](https://github.com/apache/solr/tree/main/solr/solr-ref-guide)

### CODE REVIEW

File: solr/solrj/src/java/org/apache/solr/client/solrj/impl/LBHttp2SolrClient.java [hunk 1]
```diff
@@ -301,14 +301,33 @@ public void onFailure(Throwable oe) {
   public static class Builder {
     private final Http2SolrClient http2Client;
     private final String[] baseSolrUrls;
+    private int interval;
```
Comment (dsmiley) [inline solr/solrj/src/java/org/apache/solr/client/solrj/impl/LBHttp2SolrClient.java:304]: aliveCheckInterval would be clearer!

Comment (dsmiley) [reply solr/solrj/src/java/org/apache/solr/client/solrj/impl/LBHttp2SolrClient.java:304]: On the other hand, this merely uses the very same field name used in the client so... let's either change them all or leave them all (as you wish).

File: solr/solrj/src/java/org/apache/solr/client/solrj/impl/LBHttpSolrClient.java [hunk 2]
```diff
@@ -259,6 +261,21 @@ public Builder withBaseSolrUrls(String... solrUrls) {
       return this;
     }
 
+    /**
+     * LBHttpSolrServer keeps pinging the dead servers at fixed interval to find if it is alive. Use
+     * this to set that interval
+     *
+     * @param interval time in milliseconds
+     */
+    public Builder setAliveCheckInterval(int interval) {
```
Comment (dsmiley) [inline solr/solrj/src/java/org/apache/solr/client/solrj/impl/LBHttpSolrClient.java:270]: use "with" builder naming convention

File: solr/solrj/src/java/org/apache/solr/client/solrj/impl/LBHttp2SolrClient.java [hunk 3]
```diff
@@ -301,14 +301,33 @@ public void onFailure(Throwable oe) {
...
       this.baseSolrUrls = baseSolrUrls;
     }
 
+    /**
+     * LBHttpSolrServer keeps pinging the dead servers at fixed interval to find if it is alive. Use
+     * this to set that interval
+     *
+     * @param interval time in milliseconds
+     */
+    public LBHttp2SolrClient.Builder setAliveCheckInterval(int interval) {
```
Comment (dsmiley) [inline solr/solrj/src/java/org/apache/solr/client/solrj/impl/LBHttp2SolrClient.java:317]: use "with" builder naming convention

**Review by dsmiley — CHANGES_REQUESTED** [review]
needs CHANGES.txt too

File: solr/solrj/src/java/org/apache/solr/client/solrj/impl/LBHttpSolrClient.java (hunk 2 above)
Comment (epugh) [reply solr/solrj/src/java/org/apache/solr/client/solrj/impl/LBHttpSolrClient.java:270]: I want to fix this as part of SOLR-16590 to get teh same name everywhere...

**Comment by epugh** [comment]
> needs CHANGES.txt too

Went ahad and added it.  Ugh, got conflicts.

File: solr/solrj/src/java/org/apache/solr/client/solrj/impl/LBHttpSolrClient.java [hunk 4]
```diff
@@ -181,6 +182,7 @@ public HttpClient getHttpClient() {
   public static class Builder extends SolrClientBuilder<Builder> {
     protected final List<String> baseSolrUrls;
     protected HttpSolrClient.Builder httpSolrClientBuilder;
+    private int aliveCheckInterval;
```
Comment (stillalex) [inline solr/solrj/src/java/org/apache/solr/client/solrj/impl/LBHttpSolrClient.java:185]: @epugh this should be init to the default CHECK_INTERVAL same as below https://github.com/apache/solr/pull/1217/files#diff-8a5ab47986f979f1134864e1352f6617e8d425f756018da7e0c1fa191fcc8fafR81

File: solr/solrj/src/java/org/apache/solr/client/solrj/impl/LBHttp2SolrClient.java [hunk 5]
```diff
@@ -301,14 +301,33 @@ public void onFailure(Throwable oe) {
   public static class Builder {
     private final Http2SolrClient http2Client;
     private final String[] baseSolrUrls;
+    private int aliveCheckInterval;
```
Comment (stillalex) [inline solr/solrj/src/java/org/apache/solr/client/solrj/impl/LBHttp2SolrClient.java:304]: same for this one. default should be CHECK_INTERVAL

**Comment by mtchorz** [comment]
@epugh After deprecating [setAliveCheckInterval](https://github.com/apache/solr/blob/releases/solr/9.4.0/solr/solrj/src/java/org/apache/solr/client/solrj/impl/LBSolrClient.java#L450), how I can set aliveCheckInterval when I am using CloudHttp2SolrClient ?

CloudHttp2SolrClient builder does not allow to create LBHttp2SolrClient.

**Pull request merged**